src/components/activities.js

```diff
@@ -241,6 +241,33 @@ export class ActivityPorts {
   constructor(win) {
     /** @private @const {!web-activities/activity-ports.ActivityPorts} */
     this.activityPorts_ = new WebActivityPorts(win);
+
+    /** @private {?../proto/api_messages.AnalyticsContext} */
+    this.analyticsContext_ = null;
+  }
+
+  /**
+   *
+   * @param {!../proto/api_messages.AnalyticsContext} context
+   */
+  setAnalyticsContext(context) {
+    this.analyticsContext_ = context;
+  }
+  /**
+   * Adds universal arguments
+   * @param {?Object=} opt_args
+   * @return {!Object}
+   * @private
+   */
+  getArgs_(opt_args) {
```

Comment:
Could we call this method `getArgsWithDefaults`, just to make it's purpose more obvious?

Reply:
On second thought, maybe something like `getArgsWithInvariants` would be better, since the `analyticsContext` arg isn't just a default value here, it would be overridden if it were passed in the `opt_args` object

Reply:
Now that you mention it, we should probably respect the value passed in if one is.